cpVase_w_holesSA.m

```diff
@@ -0,0 +1,94 @@
+function [cpx,cpy,cpz,dist,bdy,SA_final] = cpVase_w_holesSA(x, y, z, ...
+    lim,ab,cen,holes,tol,SA_wanted,dx,bw)
+%cpf, holes,...
+
+% x,y,z coordinate matrices of any same dimension
+% lim = [angle from xy-plane, angle from xy-plane], a range of angle.
+%   e.g. lim = [-pi/2, pi/2] makes a full ellipsoid
+
+
+% just project onto vase
+%cpf = @(x,y,z) cpVase(x, y, z, lim, ab, cen);
+%[cpx,cpy,cpz,~,~] = cpf;
+
+[cpx,cpy,cpz,~,~] = cpVase(x, y, z, lim, ab, cen);
+bdy = zeros(size(cpx));
+size_holes=size(holes);
+num_holes=size_holes(1);
+
+for k=1:num_holes
+    hole = holes(k,:);
+    
+    xh = hole(1);
+    yh = hole(2);
+    zh = hole(3);
+    Rh = hole(4);
+    
+    if exist('SA_wanted','var')
+        
+        SA_wanted_k = SA_wanted(k);
+        R_bar = sqrt(SA_wanted_k/pi);
+        RR = linspace(0.8*R_bar,1.2*R_bar, 50);
```

Comment:
BTW, maybe 0.8 is overkill: it cannot be less than the planar case (right?)  maybe `0.99*R_bar, 1.5*R*bar`.  What is a good estimate for `1.5`?  That is, how much larger than the planar case can it realistically be?